### PR TITLE
Add support for setting the `User` metadata item

### LINKED ISSUE #115: Implement user/group metadata setting
Two new string properties:
* `ContainerUser`
* `ContainerGroup`

If `ContainerGroup` is specified, `ContainerUser` _must_ be specified as well.

Will be emitted to the `User` field of the image config in the form `<user string value>` if only `ContainerUser` is specified, `<user string value>:<group string value>` if both are specified.

**References**
https://github.com/opencontainers/image-spec/blob/main/config.md

### PR DESCRIPTION
**Pull request by baronfel**
There is no validation on this, as it's _very_ hard to do - the supported formats are

* user name
* user id
* user name:group name
* user id:group id
* user id:group name
* user name:group id

all of which should match either a name or an id from `/etc/passwd`, which we theoretically have access to through the layers, but that seems _really_ extra to acquire, unTAR, parse and validate.

Closes #115

### CODE REVIEW

**Comment by baronfel** [comment]
This builds on #358 so shouldn't merge until it does.

File: Microsoft.NET.Build.Containers/Tasks/CreateNewImage.Interface.cs
Comment (vlada-shubina) [inline Microsoft.NET.Build.Containers/Tasks/CreateNewImage.Interface.cs:124]: unless I'm missing something, this should be added to `CreateNewImageToolTask` to be passed to `containerize`.

**Review by vlada-shubina — CHANGES_REQUESTED** [review]
Please check if changes to `CreateNewImageToolTask` are needed.

File: Microsoft.NET.Build.Containers/Tasks/CreateNewImage.Interface.cs
Comment (baronfel) [reply Microsoft.NET.Build.Containers/Tasks/CreateNewImage.Interface.cs:124]: Excellent catch - I added it to the exe and didn't plumb it through here.

Comment (baronfel) [reply Microsoft.NET.Build.Containers/Tasks/CreateNewImage.Interface.cs:124]: Though I think we're missing a merge/rebase in the ToolsTask on this branch - I think you put a lot of changes in there recently?

**Review by vlada-shubina — APPROVED** [review]

**Comment by baronfel** [comment]
I will rebase/patch up after #358 is merged

**Pull request merged**
